### ISSUE TITLE
How to catch the abnormal situation during battery operation by  Petlion

### ISSUE THREAD

**Issue by lico0001** [MarcBerliner/PETLION.jl#103]
Hi MarcBerliner.  I want to catch the outlier during battery operation, that is, if the temperature is set too high, the current and voltage will change abnormally during battery operation. BBut in my previous attempts, if I set it this way, the program would report an error. May I ask if this data can be obtained through Petlion? If so, is there a suitable example? Looking forward to your reply, thank you.

**Comment by MarcBerliner** [comment]
To clarify, is your simulation failing at high temperatures but you still want to receive the output?

**Comment by lico0001** [comment]
Yes, but what I want to know more is how to get the outlier through the status of the Petlion battery crash.

**Comment by MarcBerliner** [comment]
This isn't explicitly accounted for, but there is a workaround.  Say you want to run a CC-CV simulation
```julia
p = petlion(LCO)

sol = simulate(p, I=5, SOC=0)
simulate!(sol, p, V=:hold)
```
which crashes with the message

```julia
julia> ERROR: Model failed to converge at t = 101.5279999999999
```

To save the output, first initialize the solution object `sol` and enclose your simulation loop in a `try/catch` block. The solve still fails, but you can access the final states before failure.
```julia
sol = solution()

try
    simulate!(sol, p, I=5, SOC=0)
    simulate!(sol, p, V=:hold)
catch
    # input catch here
end
````
```julia
julia> PETLION simulation
 --------
 Run:     I
 Time:    485.27 s
 Current: 3.5487C
 Voltage: 4.3 V
 Power:   446.03 W/m²
 SOC:     0.6485
 Exit:    Above max. voltage
```
